system-test/storage.ts

```diff
@@ -780,6 +780,120 @@ describe('storage', () => {
     });
   });
 
+  describe('public access prevention', () => {
+    let bucket: Bucket;
+
+    const PUBLIC_ACCESS_PREVENTION_UNSPECIFIED = 'unspecified';
+    const PUBLIC_ACCESS_PREVENTION_ENFORCED = 'enforced';
+
+    const createBucket = () => {
+      bucket = storage.bucket(generateName());
+      return bucket.create();
+    };
+
+    const setPublicAccessPrevention = (bucket: Bucket, configuration: string) =>
+      bucket.setMetadata({
+        iamConfiguration: {
+          publicAccessPrevention: configuration,
+        },
+      });
+
+    const validateUnexpectedPublicAccessPreventionValueError = (
+      err: ApiError
+    ) => {
+      assert.strictEqual(err.code, 400);
+      return true;
+    };
+
+    const validateConfiguringPublicAccessWhenPAPEnforcedError = (
+      err: ApiError
+    ) => {
+      assert.strictEqual(err.code, 412);
+      return true;
+    };
+
+    before(createBucket);
+
+    it('inserts a bucket with enforced public access prevention', async () => {
+      await setPublicAccessPrevention(
+        bucket,
+        PUBLIC_ACCESS_PREVENTION_ENFORCED
+      );
+      const [bucketMetadata] = await bucket.getMetadata();
+      const publicAccessPreventionStatus =
+        bucketMetadata.iamConfiguration.publicAccessPrevention;
+      await assert.rejects(
+        () => bucket.makePublic(),
```

Comment:
Perhaps should also check that it rejects an object being made public via ACLs, or a set of objects via IAM policy for a prefix.